python/lsst/meas/algorithms/gp_interpolation.py

```diff
@@ -0,0 +1,657 @@
+from lsst.geom import Box2I, Point2I, Extent2I
+from lsst.afw.geom import SpanSet
+import copy
+import treegp
+
+import jax
+from jax import jit
+import jax.numpy as jnp
+
+import warnings
+
+__all__ = ["interpolateOverDefectsGP"]
+
+
+def updateMaskFromArray(mask, bad_pixel, interpBit):
+    """
+    Update the mask array with the given bad pixels.
+
+    Parameters
+    ----------
+    mask : MaskedImage #TO DO, this is not the right format need to check.
+        The mask image to update.
+    bad_pixel : array-like
+        An array-like object containing the coordinates of the bad pixels.
+        Each row should contain the x and y coordinates of a bad pixel.
+    interpBit : int
+        The bit value to set for the bad pixels in the mask.
+
+    Returns
+    -------
+    None
+    """
+    x0 = mask.getX0()
+    y0 = mask.getY0()
+    for row in bad_pixel:
+        x = int(row[0] - x0)
+        y = int(row[1] - y0)
+        mask.array[y, x] |= interpBit
+
+
+@jit
+def median_with_mad_clipping(data, mad_multiplier=2.0):
+    """
+    Calculate the median of the input data after applying Median Absolute Deviation (MAD) clipping.
+
+    Parameters:
+    -----------
+    data : array-like
+        Input data array.
+    mad_multiplier : float, optional
+        Multiplier for the MAD value used for clipping. Default is 2.0.
+
+    Returns:
+    --------
+    median_clipped : float
+        Median value of the clipped data.
+
+    Notes:
+    ------
+    The MAD clipping method is used to remove outliers from the data. The median of the data is calculated,
+    and then the MAD is calculated as the median absolute deviation from the median. The data is then clipped
+    by removing values that are outside the range of median +/- mad_multiplier * MAD. Finally, the median of
+    the clipped data is returned.
+
+    Examples:
+    ---------
+    >>> data = [1, 2, 3, 4, 5, 100]
+    >>> median_with_mad_clipping(data)
+    3.5
+    """
+    median = jnp.median(data)
+    mad = jnp.median(jnp.abs(data - median))
+    clipping_range = mad_multiplier * mad
+    clipped_data = jnp.clip(data, median - clipping_range, median + clipping_range)
+    median_clipped = jnp.median(clipped_data)
+    return median_clipped
+
+
+# Below are the jax functions for Gaussian Process regression.
+# Kernel is assumed to be isotropic RBF kernel, and solved
+# using exact Cholesky decomposition.
+# The interpolation solution is obtained by solving the linear system:
+# y_interp = kernel_rect @ (kernel + y_err**2 * I)^-1 @ y_training.
+# See the Rasmussen and Williams book for more details.
+# Each function is decorated with @jit to compile the function.
+# Exist package like tinygp, that is implemented in jax also, but
+# wanted to control the implementation here, as I am doing a lot
+# of custom things (setting my own hyperparameters, fine tune mean function,
+# dynamic binning, ...).
+
+
+@jit
```

Comment:
If I put them like this, this is because I was not able to make it work within class and there is not an obvious / fast way to make it work within class I was able to found.